cc_macro.c

```diff
@@ -424,6 +485,44 @@ int macro_expression(void)
 	return macro_bitwise_expr();
 }
 
+void handle_function_like_macro(struct macro_list* hold)
+{
+
+	eat_current_token(); /* Skip '(' */
+	struct macro_argument* argument = calloc(1, sizeof(struct macro_argument));
+	hold->arguments = argument;
+
+	argument->next = NULL;
+	/* A NULL name in the first argument means a function macro without any arguments */
+	argument->name = NULL;
+
+	while(macro_token->s[0] != ')')
+	{
+		/* ... is not tokenized as a single token here */
+		if(macro_token->s[0] == '.' &&
+			macro_token->next->s[0] == '.' &&
```

Comment:
We need to add more error checking here and in other similar places where pointers might be NULL. E.g. something like

```
#define A(x, y) x+y

int main() {
  A(x, 
  return 0;
}
```
would cause a crash.

Reply:
Fixed.
I updated it to use the `require` function although it doesn't print file and line info which is very useful.